contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -437,7 +437,7 @@ func (q *queuer) Run(done <-chan struct{}) {
 			}
 
 			pod := p.(*Pod)
-			if recoverAssignedSlave(pod.Pod) != "" {
+			if recoverAssignedSlave(pod.Pod) != "" || pod.InGracefulTermination() {
```

Comment:
I agree. The "running" check must go for my reasoning above.

Moreover, I agree that we are really talking about some corner case:
- pre-scheduled
- gracefully deleted before being scheduler.

In the worst case the pod is 
- either never launched because the node does not exist. Then it stays in the scheduling queue with some backoff between each scheduling run, until the grace period is over
- or it is launched and directly terminated by the kubelet.

What's your suggestion about this? Drop it or keep it?